cwltool/docker.py

```diff
@@ -236,7 +236,7 @@ def add_file_or_directory_volume(
         """Append volume a file/dir mapping to the runtime option list."""
         if not volume.resolved.startswith("_:"):
             _check_docker_machine_path(volume.resolved)
-            self.append_volume(runtime, volume.resolved, volume.target)
+            self.append_volume(runtime, volume.resolved, volume.target, writable=volume.staged)
```

Comment:
What's the goal here? This results in many files that should be mounted read-only to not be..

Reply:
The goal overall was to fix a bug I was encountering with a specific CommandLineTool (Yank) which we are no longer using. I still believe there is a bug in cwltool, but I can't seem to simultaneously fix the bug and get the test suite to pass. This is very low priority work for me, so feel free to close if you like.